enhancements/microshift/audit-log-configuration-options.md

```diff
@@ -0,0 +1,245 @@
+
+### Goals
+
+- Enable MicroShift administrators to manage logging policies as a set of hierarchical profiles.
+- Provide flexibility to specify log file sizes and retention policies.
+
+### Non-Goals
+
+- Custom Rules: MicroShift is a single-user system without user groups, so custom audit log rules similar to OpenShift are not 
+required and should be explicitly marked as out of scope.
+- Support OVN-K audit log configuration. OVN-K audit log policies are managed via a configMap and do not have the same flexibility a OpenShift and Kubernetes API servers.
+
+## Proposal
+
+This proposes exposing a subset of kube-apiserver audit-log settings flags. The kube-apiserver settings will enable user control over log file rotation and retention. Users may set fields in combination to define a maximum storage limit (e.g. max num files * max single file size = total storage limit). This is a critical feature for far edge devices with limited storage capacities. On such devices, logging data accumulation risks starving the host system or cluster workloads, potentially bricking the device until human intervention can be applied.  Thus, it is necessary to provide users a means of enforcing such a limit and what actions to take at that limit. Users must also be able to select which events are logged. This will be exposed as a set of "profiles," with predefined behaviours and will be mutually exclusive. This should give users holistic control of their audit log rotation, retention, and max allowable storage allocation.
+
+### Workflow Description
+
+1. Administrator edits MicroShift config file to specify desired audit logging policy profile
+2. Administrator edits MicroShift config file to specify max file size, number of files total, and max age of files for logs
+3. MicroShift service is restarted to apply changes
+
+### API Extensions
+
+**Audit Log Policy Configuration:**
+
+- MicroShift config’s `apiServer` root-field must include a child node for storing log configurations.
+
+```yaml
+apiServer:
+  auditLog: MAP(STRING)INTERFACE
+```
+
+- The `apiServer.auditLog` field must include a child node for storing the audit log policy profile to use. See [Implementation Details](#implementation-details) for more. These are the same policies are equal to [Openshift Audit log policies](https://docs.openshift.com/container-platform/4.14/security/audit-log-policy-config.html#about-audit-log-profiles_audit-log-policy-config).
+
+```yaml
+apiServer:
+  auditLog:
+    policy: STRING
+```
+
+**Audit Log File Rotation:**
+
+- `maxFileSize` config variable to set maximum audit log file size. Accepted values must be a string with a number and unit suffix with no separator character, e.g. `10GB`. Valid units are "KB", "MB", "GB", "TB", "KiB", "MiB", "GiB", and "TiB". Since the kube-apiserver only accepts max file size defined in bytes, MicroShift will need to translate this field value internally.
```

Comment:
> Since the kube-apiserver only accepts max file size defined in **bytes**

From https://kubernetes.io/docs/reference/command-line-tools-reference/kube-apiserver/:

> --audit-log-maxsize int
>   The maximum size in **megabytes** of the audit log file before it gets rotated.

Reply:
Good catch! I'd read it several times but it hadn't quite stuck. Will adjust this accordingly.